File: Logic & Design/Visual Studio Code/Menu-Driven Programming/Java/Algorithm1.java

/**Documentation
	Algorithm 1
	Programmer: Corey Jenkins
	Date: 11-11-2017
**/

import java.util.Scanner;

public class Algorithm1 {
	public static void main(String[] args) {
		Menu();
	}

	public static void Menu() {
		int menuSelection;
		Scanner keyboard = new Scanner(System.in);
		System.out.print("This is the main menu.");

		// The menu
		System.out.println("1. Open a menu");
		System.out.println("2. Close the currrent document");
		System.out.println("3. Print the current document.");
		System.out.println("4. Exit the program.");
		System.out.println();

		System.out.println("Enter your selection.");
		menuSelection = keyboard.nextInt();

		// Peform the selected operation
		if (menuSelection == 1)
			System.out.println("Opening the document.");
		else if (menuSelection == 2)
			System.out.println("Closing the current document.");
		else if (menuSelection == 3)
			System.out.println("Printing the current document.");
		else if (menuSelection == 4)
			System.out.println("Exiting the program.");
		else
			System.out.println("That is an invalid selection.");
	}
}

File: Logic & Design/Visual Studio Code/Menu-Driven Programming/Python/algorithm1.py
'''
Programmer:     Corey Jenkins
Date:           January 24, 2023
Filename:       algorithm1.py
Purpose:        An algorithm that is an event-driven program.
'''

def main():
    menu()
    
def menu():
    print("This is the main menu.")
    print("1. Open a menu.")
    print("2. Close the current document.")
    print("3. Print the current document.")
    print('4. Exit the program.')
    menu_selection = int(input("Enter your selection: "))
    
   # Python does not have case/switch; therefore, if/else statements must be used.
    
    if (menu_selection == 1):
        print("Opening the menu.")
    elif (menu_selection == 2):
        print('Closing the document.')
    elif (menu_selection == 3):
        print("Printing the document.")
    elif (menu_selection == 4): 
        print("Exiting the program.")
    else:
        print("You have made an invalid seleciton.")
menu()
        